OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -96,6 +119,103 @@ public void onPageLoadProgressChanged(WebView webView, int progress) {
         }
     }
 
+    private void fetchDateBannerInfo() {
+        if (canFetchBannerInfo && unit.getType() == BlockType.PROBLEM) {
+            courseDateViewModel = new ViewModelProvider(this, new ViewModelFactory()).get(CourseDateViewModel.class);
+            initObserver();
```

Comment:
`initObserver` should be move-in `onViewCreated` and call only once. that also remove the recursive behavior of webview loading.

Reply:
Done